src/buckets/components/createBucketForm/BucketOverlayForm.tsx

```diff
@@ -18,8 +18,17 @@ import {
 import {RuleType} from 'src/buckets/reducers/createBucket'
 import {isFlagEnabled} from 'src/shared/utils/featureFlag'
 import {CLOUD} from 'src/shared/constants'
-import {SchemaType} from 'src/client/generatedRoutes'
 
+import {MeasurementSchemaSection} from './MeasurementSchemaSection'
+
+let MeasurementSchemaList = null,
+  SchemaType = null
+
+if (CLOUD) {
+  SchemaType = require('src/client/generatedRoutes').MeasurementSchema
+  MeasurementSchemaList = require('src/client/generatedRoutes')
+    .MeasurementSchemaList
+}
```

Comment:
where did this pattern come from and is anyone aware if it's actually doing anything beneficial? it looks like it's there as a performance thing to not load components that aren't used (assumptions about intent are dangerous), but is that actually having any measurable effect on bundle size, app performance or feel of the app? did anyone check check up on any of this before putting this in the codebase?

it's definitely making the code more complex and as we've already seen, is a source of catastrophic errors.

Reply:
if it's not there, then the eslint when it is *not* cloud fails b/c those calls are only available in cloud.

i'm using the same exact pattern as in my earlier commits in this epic.  

i copied the pattern from elsewhere ; if there's a better pattern you know of please let me know and I can switch to it

Reply:
see ui/src/cloud/actions/limits.ts

Reply:
thanks for the information. So this is less a performance pattern and more to get around fundamental issues in the architecture of the app. The first is that our app doesn't make any architectural distinction between oss and cloud. All the distinctions are made at code levels with branching `if (CLOUD)` statements. The second is that openapi _does_ make a distinction between cloud and OSS types, and we have to account for that with our lack of architectural distinction